### PR TITLE
[FIX] account_payment_term_discount

### PR DESCRIPTION
**Pull request by max3903**

### CODE REVIEW

**Comment by max3903** [comment]
Ping @Hardik-OSI

**Review by bodedra — APPROVED** [review]
LG :+1:

File: account_payment_term_discount/wizard/account_payment_register.py [hunk 1]
```diff
@@ -17,14 +17,14 @@ class AccountPaymentRegister(models.TransientModel):
 
     @api.model
     def default_get(self, fields):
-        res = super(AccountPaymentRegister, self).default_get(fields)
+        res = super().default_get(fields)
         # invoice_defaults = self.resolve_2many_commands('invoice_id',
         #                                                res.get('invoice_id'))
         active_id = self.env.context.get("active_ids", [])
```
Comment (max3903) [inline account_payment_term_discount/wizard/account_payment_register.py:21]: @bodedra Are you sure `active_ids` contains ID of `account.move`?

I get the following error running tests:

```
  File "/odoo/addons/account_payment_term_discount/wizard/account_payment_register.py", line 27, in default_get
    res["discount_amt"] = record.discount_amt or 0
  File "/usr/lib/python3/dist-packages/odoo/fields.py", line 1000, in __get__
    raise MissingError("\n".join([
odoo.exceptions.MissingError: Record does not exist or has been deleted.
(Record: account.move(1533,), User: 568)
```

Comment (bodedra) [reply account_payment_term_discount/wizard/account_payment_register.py:21]: @max3903 Nope. active_ids don't contains `account.move`. For that, we need to iterate record and use `invoice_id` field.

Comment (max3903) [reply account_payment_term_discount/wizard/account_payment_register.py:21]:
```
2021-07-23 21:55:21,063 1 INFO FMK odoo.addons.account.tests.test_account_bank_statement: Starting TestAccountBankStatementLine.test_reconciliation_statement_line_with_generated_payments ... 
> /odoo/addons/account_payment_term_discount/wizard/account_payment_register.py(24)default_get()
-> active_id = self.env.context.get("active_ids", [])
(Pdb) self.env.context
{'active_model': 'account.move.line', 'active_ids': [1533]}
(Pdb) n
> /odoo/addons/account_payment_term_discount/wizard/account_payment_register.py(25)default_get()
-> record = self.env["account.move"].browse(active_id)
(Pdb) n
> /odoo/addons/account_payment_term_discount/wizard/account_payment_register.py(26)default_get()
-> if record and len(record) == 1:
(Pdb) record.discount_amt
*** odoo.exceptions.MissingError: Record does not exist or has been deleted.
(Record: account.move(1533,), User: 568)
```

Comment (bodedra) [reply account_payment_term_discount/wizard/account_payment_register.py:21]: @max3903 Thanks for your debug output. I updated code here directly. would you please test it.

Comment (bodedra) [reply account_payment_term_discount/wizard/account_payment_register.py:21]: @max3903 I need to review it and resolve this issue.

Comment (RLeeGML) [reply account_payment_term_discount/wizard/account_payment_register.py:21]: @bodedra  I put in a fix that should work if the active_model is account.move, account.move.line, or neither

**Review by RLeeGML — APPROVED** [review]
Checks passed, LGTM

File: account_payment_term_discount/wizard/account_payment_register.py (hunk 1 above)
Comment (bodedra) [reply account_payment_term_discount/wizard/account_payment_register.py:21]: That's great. Thanks @RLeeOSI

**Pull request merged**